internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -921,6 +908,43 @@ func (t *Translator) buildListenerTLSParameters(policy *egv1a1.ClientTrafficPoli
 	return irTLSConfig, nil
 }
 
+// validateAndGetDataAtKeyInRef validates the secret object reference and gets the data at the key in the secret or configmap
+func (t *Translator) validateAndGetDataAtKeyInRef(ref gwapiv1.SecretObjectReference, key string, resources *resource.Resources, from crossNamespaceFrom) ([]byte, error) {
+	refKind := string(ptr.Deref(ref.Kind, resource.KindSecret))
+	switch refKind {
+	case resource.KindSecret:
+		secret, err := t.validateSecretRef(false, from, ref, resources)
+		if err != nil {
+			return nil, err
+		}
+
+		secretCertBytes, ok := getOrFirstFromData(secret.Data, key)
+		if !ok || len(secretCertBytes) == 0 {
+			return nil, fmt.Errorf("ref secret [%s] not found", ref.Name)
```

Comment:
Nit: this error could cause confusion when the secret exists but data is nil, could we change it to:
```suggestion
			return nil, fmt.Errorf("ref secret [%s] missing data for key %q", ref.Name, key)
```